fastestimator/estimator.py

```diff
@@ -132,6 +140,10 @@ def _start(self):
             pass  # On early stopping we still want to run the final traces and return results
         self._run_traces_on_end()
 
+    def _start_test(self):
```

Comment:
this misses run_traces_on_begin and run_traces_on_end which may be required for test traces too if setup / teardown occurs there

Reply:
`eval` doesn't run it, therefore, `test` shouldn't run it

Reply:
start and end are not mode dependent. eval technically does run it

Reply:
sure. I will add it there too.

Reply:
do we also need to call _prepare_system() and then set those variables appropriately here? what would the appropriate value for current epoch even be in this case?

Reply:
i think we still need to call _prepare_system() here too - suppose user builds an estimator and calls .test() without first calling .fit(). the traces would then not have received the system variable yet right?